src/Compilers/Core/CodeAnalysisTest/Diagnostics/CompilationWithAnalyzersTests.cs

```diff
@@ -74,7 +74,9 @@ public void TestIsDiagnosticAnalyzerSuppressedWithExceptionInSupportedDiagnostic
         {
             // Verify IsDiagnosticAnalyzerSuppressed does not throw an exception when 'onAnalyzerException' is null.
             var analyzer = new AnalyzerThatThrowsInSupportedDiagnostics();
+#pragma warning disable CS0618 // Type or member is obsolete
```

Comment:
💡 This is a case where it seems like it would be easier to mark the test obsolete with a comment saying the test covers obsolete public API functionality